### PR TITLE
Change config unstable/deprecations warnings to use the aggregated config warnings

### LINKED ISSUE #1860: mirrord config - change unstable/deprecations to happen once
When we load a `LayerConfig`, if users uses unstable/deprecated feature it alerts each time we load `LayerConfig` which creates a bit of spammy behavior.
There's `verify` method for config, so we should move the logic to happen there.

### PR DESCRIPTION
**Pull request by aviramha**
Change config unstable/deprecations warnings to use the aggregated config warnings
Closes #1860

### CODE REVIEW

File: mirrord/cli/src/operator.rs [hunk 1]
```diff
@@ -71,7 +71,8 @@ async fn operator_setup(
 
 async fn get_status_api(config: Option<String>) -> Result<Api<MirrordOperatorCrd>> {
     let kube_api = if let Some(config_path) = config {
-        let config = LayerFileConfig::from_path(config_path)?.generate_config()?;
+        let mut warnings = Vec::new();
+        let config = LayerFileConfig::from_path(config_path)?.generate_config(&mut warnings)?;
```
Comment (DmitryDodzin) [inline mirrord/cli/src/operator.rs:75]: Maybe add add print for warnings if they happen?

File: mirrord/config/src/lib.rs [hunk 2]
```diff
@@ -278,12 +278,24 @@ pub struct LayerConfig {
 }
 
 impl LayerConfig {
-    pub fn from_env() -> Result<Self, ConfigError> {
+    /// Generate a config from the environment variables and/or a config file.
+    /// On success, returns the config and a vec of warnings.
+    /// To be used from CLI to verify config and print warnings
+    pub fn from_env_with_warnings() -> Result<(Self, Vec<String>), ConfigError> {
+        let mut warnings = Vec::new();
```
Comment (DmitryDodzin) [inline mirrord/config/src/lib.rs:285]: For consistency with other APIs and `from_env` prettier
```suggestion
    pub fn from_env_with_warnings(warnings: &mut Vec<String>) -> Result<(Self, Vec<String>), ConfigError> {
```

File: mirrord/config/src/util.rs [hunk 3]
```diff
@@ -35,11 +35,11 @@ where
 {
     type Generated = T::Generated;
 
-    fn generate_config(self) -> Result<Self::Generated, ConfigError> {
+    fn generate_config(self, warnings: &mut Vec<String>) -> Result<Self::Generated, ConfigError> {
```
Comment (DmitryDodzin) [inline mirrord/config/src/util.rs:38]: One little thing that will make future work easier on it is maybe wrapping the `Vec<String>` in some `ConfigContext` struct (like if maybe `String` should be `LogMessage` or have some extra data like a pretty error)

File: mirrord/cli/src/operator.rs (hunk 1 above)
Comment (aviramha) [reply mirrord/cli/src/operator.rs:75]: don't think we need it in this context

File: mirrord/config/src/util.rs (hunk 3 above)
Comment (aviramha) [reply mirrord/config/src/util.rs:38]: Nice!

File: mirrord/config/src/lib.rs (hunk 2 above)
Comment (aviramha) [reply mirrord/config/src/lib.rs]: I think the internal API shouldn't be consistent with external tbh.. also makes funky code from caller.

File: mirrord/cli/src/extension.rs [hunk 4]
```diff
@@ -57,10 +57,13 @@ pub(crate) async fn extension_exec(args: ExtensionExecArgs) -> Result<()> {
         std::env::set_var("MIRRORD_IMPERSONATED_TARGET", target.clone());
         env.insert("MIRRORD_IMPERSONATED_TARGET".into(), target.to_string());
     }
-    let config = LayerConfig::from_env()?;
+    let (config, mut context) = LayerConfig::from_env_with_warnings()?;
 
     let mut analytics = AnalyticsReporter::only_error(config.telemetry);
 
+    let mut warnings = config.verify()?;
+    warnings.extend(context.get_warnings().iter().cloned());
```
Comment (DmitryDodzin) [inline mirrord/cli/src/extension.rs:65]: add the loop with `progress.warning(&warning);` =]

**Review by DmitryDodzin — APPROVED** [review]

**Pull request merged**